docs/proposals/role-entry-points.md

```diff
@@ -0,0 +1,302 @@
+# Entry-point based roles
+
+## Variables inside a role are too open
+
+Each role consists of a set of tasks and variables read/set in some of the tasks.
+Some of the tasks can store their results into registered variables,
+e.g. store stdout of a docker pull command.
+Some of the tasks may be run only under certain conditions that are generated
+outside of a role, e.g. pull images only if a deployment is containerized.
+Some of the tasks are run only if a certain variable is defined,
+e.g. add additional Kubelet arguments into the `node-config.yml`
+if `kubelet_additional_args` inventory variable is defined.
+
+Some roles work with a lot of variables that from a point of view of a role are uninvited citizens.
```

Comment:
Anything that makes role maintenance harder. Like a global variables inside a C function, or like a variable over which I have only a limited knowledge (e.g. don't know which values are accepted, if the variable is always defined or just sometimes, or where is the variable set.